conda/_vendor/auxlib/__init__.py

```diff
@@ -45,3 +45,30 @@ def emit(self, record):
 __license__ = "ISC"
 __copyright__ = "(c) 2015 Kale Franz. All rights reserved."
 __summary__ = """auxiliary library to the python standard library"""
+
+
+class _Null(object):
+    """
+    Examples:
+        >>> len(_Null())
+        0
+        >>> bool(_Null())
+        False
+        >>> _Null().__nonzero__()
+        False
+    """
+    def __nonzero__(self):
+        return self.__bool__()
+
+    def __bool__(self):
+        return False
+
+    def __len__(self):
+        return 0
+
+
+# Use this NULL object when needing to distinguish a value from None
+# For example, when parsing json, you may need to determine if a json key was given and set
+#   to null, or the key didn't exist at all.  There could be a bit of potential confusion here,
+#   because in python null == None, while here I'm defining NULL to mean 'not defined'.
+NULL = _Null()
```

Comment:
Can we keep this named NULL for now?  And then change it in a future PR?  It's not new code, and if I rename it here, it's going to create unpleasant merge conflicts with another branch I'm working on (docs-wip-2).